backend/app/anti_recommenders/arkg/arkg_anti_recommender.py

```diff
@@ -191,7 +142,7 @@ def generate_anti_recommendations(
                 )
 
                 try:
-                    self.__upsert_user_state_into_database()
+                    upsert_user_state_into_database(self.__user_state)
```

Comment:
Per discussion, this will go away. The code here shouldn't care whether the user is in the database or not. It should call polymorphic methods on the user object to store state.

Reply:
All database calls have been removed from the `AntiRecommenders`, and a `UserService` has been added to handle a user's state management.